EXPLAINER_BLUETOOTH.md

```diff
@@ -0,0 +1,227 @@
+for Bluetooth serial ports.
+
+## Non-standard Service Class ID’s
+
+Devices supporting a Serial Port Profile
+([SPP](https://www.bluetooth.com/specifications/specs/serial-port-profile-1-2/))
+often offer this via the standard Serial Port service (UUID: `0x1101`).
+Operating systems may create devices or ports mapped to this Bluetooth service
+during the bonding process. On macOS and Windows, two device nodes are created
+for each bonded device that exposes a serial port using the standard Serial Port
+service.
+
+This proposal recommends supporting **any** service that implements the Serial
+Port Profile, regardless of whether the service is a standard Serial Port
+service or is automatically mapped by the operating system. By default
+`Serial.requestPort()` will only include Bluetooth serial ports exposed by
+devices with a standard Serial Port service. A new service UUID filter option is
+added to `Serial.requestPort()` to enable requesting serial ports for other
+services.
+
+## Potential specification changes
+
+### Port Selection
+
+[`Serial.requestPort()`](https://developer.mozilla.org/en-US/docs/Web/API/Serial/requestPort),
+with no filters parameter, will pass through:
+
+1. Non-Bluetooth serial ports (wired, virtual, etc.)
+2. Mapped Bluetooth serial ports.
+3. Any unmapped serial port provided by a standard Bluetooth SerialPort service.
+   These will appear on operating systems without automatic port mapping.
+
+### Service Class ID Filtering
+
+Serial ports for services other than the SerialPort service (UUID: `0x1101`)
+ can be requested, but will be omitted unless explicitly filtered for.
+
+`Serial.requestPort()` will be extended to add new filtering abilities for
+bonded Bluetooth devices with services implementing the SPP profile. At present
+`requestPort()` accepts a single optional
+[filters parameter](https://wicg.github.io/serial/#dom-serialportrequestoptions-filters)
+– a sequence of
+[`SerialPortFilter`](https://wicg.github.io/serial/#serialportfilter-dictionary)
+instances with the following two members:
+
+1. `usbVendorId`
+2. `usbProductId`
+
+To provide the ability to filter Bluetooth serial ports filters will need to
+support filtering for a Bluetooth service identified by a service class ID. This
+will be implemented by adding the following SerialPortFilter member:
+
+1. `bluetoothServiceClassId`
+
+Browser vendors may provide UI for the user to select certain device types such
+as Wired, Bluetooth. This is an implementation issue left to browser vendors.
+
+### Serial Port Info
+
+Currently
+[`SerialPort.getInfo()`](https://developer.mozilla.org/en-US/docs/Web/API/SerialPort/getInfo)
+returns a [`SerialPortInfo`](https://wicg.github.io/serial/#serialportinfo-dictionary)
+dictionary containing usbVendorId and usbProductId.
+
+SerialPortInfo will be extended by adding the following member:
+
+1. `bluetoothServiceClassId` (string) – The Bluetooth service class UUID. This may
+   be either a 16 or 128 bit UUID.
```

Comment:
We should probably normalize this to the 128-bit UUID form as Web Bluetooth does.

Reply:
done.